lib/prefix.c

```diff
@@ -407,6 +407,18 @@ int prefix_match(const struct prefix *n, const struct prefix *p)
 	int shift;
 	const u_char *np, *pp;
 
+	if (n->family == AF_ETHERNET)
+	{
+		if (is_zero_mac(&(n->u.prefix_eth)))
```

Comment:
is this really needed? we can just set the prefix length of a zero mac address to 0 further up, can't we?

also, is it possible the user may want to match on a partial MAC, e.g. vendor ID only?